File: authserver.js
const express     = require('express');
const app         = express();
const bodyParser  = require('body-parser');
const jwt         = require('jsonwebtoken');
const request     = require('request');
const jwkToPem    = require('jwk-to-pem');

const port = process.env.PORT || 8080; 
app.set('superSecret', 'NUP5MEqlZQBOBkJu/+twgp4tSJmSajqsIzA3MCMDzog='); 

app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());

// basic route
app.get('/', function(req, res) {
    res.send('Hello! The API is at http://localhost:' + port + '/api');
});

app.post('/oauth2/auth', function(req, res, next){
    const token = req.headers["authorization"].replace(/bearer /i,'');
    const decoded = jwt.decode(token, {complete: true});
    
    function extractKeys(error, response, body) {
        if(error)
            throw new Error(error);
    
        const jwk_keys = JSON.parse(response.body);
        let keys = jwk_keys.keys.filter((x) => x['kid'] === decoded.header['kid']);
        
        if (keys.length)
            validateToken(keys[0]);
    }

    function validateToken(key) {
        console.log(decoded.header);
        const pem = jwkToPem(key);

        jwt.verify(token, pem, function(err, decoded) {
            if(err)
                res.status(403).send(err);
            
            return(res.status(200).send('OK'));
        })
    }
    request('https://login.windows.net/common/discovery/keys',extractKeys)

});


app.listen(port);
console.log('listening on http://localhost:' + port);

File: authserver.py
from bottle import route, run, template, get, post, request, redirect # pip install bottle
import jwt  # pip install pyjwt cryptography requests
import requests as req

@route('/hello/<name>')
def index(name):
    return template('<b>Hello {{name}}</b>!', name=name)

@post('/oauth2/auth') # or @route('/login', method='POST')
def auth():
    if authenticated():        
        return "AUTH OK"
    else:
        return "<p>Not Authenticated</p>"

def authenticated():
    auth_header = request.headers.get('Authorization')
    print(auth_header)

    app_id = 'daa19535-fa14-4b4c-9cd4-d5511fcb9273'
    access_token = auth_header.replace('Bearer ','')
    token_header = jwt.get_unverified_header(access_token)

    #res = req.get('https://login.microsoftonline.com/a84894e7-87c5-40e3-9783-320d0334b3cc/.well-known/openid-configuration')
    #jwk_uri = res.json()['jwks_uri']
    #print(jwk_uri)
    return True

run(host='0.0.0.0', port=8080)